### PR TITLE
Updates for docker compose plugin.

### PR DESCRIPTION
**Pull request by jvanderaa**
# Closes: #191

## What's Changed

Moves from docker-compose standalone to docker compose plugin

## TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/chatops/en/latest/dev/dev_contributing/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: docs/dev/dev_environment.md [hunk 1]
```diff
@@ -73,7 +76,7 @@ nautobot-server migrate
 ```
 
 !!! note
-    If you want to develop on the latest develop branch of Nautobot, run the following command: `poetry add --optional git+https://github.com/nautobot/nautobot@develop`. After the `@` symbol must match either a branch or a tag.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:76]: Can you revert these formatting changes? The MKDocs admonitions we use for these requires the indentation.

File: docs/dev/dev_environment.md [hunk 2]
```diff
@@ -25,7 +25,10 @@ Using **Invoke** these configuration options can be overridden using [several me
 ### Docker Development Environment
 
 !!! tip
-    This is the recommended option for development.
+As of 1.11.1 the Docker Compose environment has moved from the standalone docker-compose to the plugin execution of `docker compose`. Be sure that your version of Docker has the plugin installed. If the docker compose plugin is installed, no other changes to the invoke methodology is required.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:28]: Add indentation for MKDocs admonition.

File: docs/dev/dev_environment.md [hunk 3]
```diff
@@ -102,7 +105,7 @@ The project features a CLI helper based on [Invoke](https://www.pyinvoke.org/) t
 Each command can be executed with `invoke <command>`. All commands support the arguments `--nautobot-ver` and `--python-ver` if you want to manually define the version of Python and Nautobot to use. Each command also has its own help `invoke <command> --help`
 
 !!! note
-    To run the mysql (mariadb) development environment, set the environment variable as such `export NAUTOBOT_USE_MYSQL=1`.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:105]: Revert

File: docs/dev/dev_environment.md [hunk 4]
```diff
@@ -136,15 +139,14 @@ Each command can be executed with `invoke <command>`. All commands support the a
...
 
 The upside to having the Nautobot service handled by Docker rather than locally is that you do not have to manage the Nautobot server. The [Docker logs](#docker-logs) provide the majority of the information you will need to help troubleshoot, while getting started quickly and not requiring you to perform several manual steps and remembering to have the Nautobot server running in a separate terminal while you develop.
 
 !!! note
-	The local environment still uses Docker containers for the supporting services (Postgres, Redis, and RQ Worker), but the Nautobot server is handled locally by you, the developer.
+The local environment still uses Docker containers for the supporting services (Postgres, Redis, and RQ Worker), but the Nautobot server is handled locally by you, the developer.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:149]: Add Indentation for MKDocs admonition.

File: docs/dev/dev_environment.md [hunk 5]
```diff
@@ -229,7 +231,7 @@ Once the containers are fully up, you should be able to open up a web browser, a
 - A live version of the documentation at [http://localhost:8001](http://localhost:8001)
 
 !!! note
-	Sometimes the containers take a minute to fully spin up. If the page doesn't load right away, wait a minute and try again.
+Sometimes the containers take a minute to fully spin up. If the page doesn't load right away, wait a minute and try again.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:234]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 6]
```diff
@@ -240,7 +242,7 @@ The Nautobot development image will automatically provision a super user when sp
 - `NAUTOBOT_SUPERUSER_PASSWORD=admin`
 
 !!! note
-	The default username is **admin**, but can be overridden by specifying **NAUTOBOT_SUPERUSER_USERNAME**.
+The default username is **admin**, but can be overridden by specifying **NAUTOBOT_SUPERUSER_USERNAME**.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:245]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 7]
```diff
@@ -290,7 +292,7 @@ Removing network nautobot_chatops_default
 This will safely shut down all of your running Docker containers for this project. When you are ready to spin containers back up, it is as simple as running `invoke start` again [as seen previously](#invoke-starting-the-development-environment).
 
 !!! warning
-	If you're wanting to reset the database and configuration settings, you can use the `invoke destroy` command, but **you will lose any data stored in those containers**, so make sure that is what you want to do.
+If you're wanting to reset the database and configuration settings, you can use the `invoke destroy` command, but **you will lose any data stored in those containers**, so make sure that is what you want to do.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:295]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 8]
```diff
@@ -301,12 +303,12 @@ Now you can start developing your plugin in the project folder!
 The magic here is the root directory is mounted inside your Docker containers when built and ran, so **any** changes made to the files in here are directly updated to the Nautobot plugin code running in Docker. This means that as you modify the code in your plugin folder, the changes will be instantly updated in Nautobot.
 
 !!! warning
-	There are a few exceptions to this, as outlined in the section [To Rebuild or Not To Rebuild](#to-rebuild-or-not-to-rebuild).
+There are a few exceptions to this, as outlined in the section [To Rebuild or Not To Rebuild](#to-rebuild-or-not-to-rebuild).
```
Comment (smk4664) [inline docs/dev/dev_environment.md:306]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 9]
```diff
@@ -301,12 +303,12 @@ Now you can start developing your plugin in the project folder!
...
 
 The back-end Django process is setup to automatically reload itself (it only takes a couple of seconds) every time a file is updated (saved). So for example, if you were to update one of the files like `tables.py`, then save it, the changes will be visible right away in the web browser!
 
 !!! note
-	You may get connection refused while Django reloads, but it should be refreshed fairly quickly.
+You may get connection refused while Django reloads, but it should be refreshed fairly quickly.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:311]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 10]
```diff
@@ -382,7 +384,7 @@ Before you continue, you'll need to update the file `development/nautobot_config
 Once the containers are up and running, you should now see the new plugin installed in your Nautobot instance.
 
 !!! note
-    You can even launch an `ngrok` service locally on your laptop, pointing to port 8080 (such as for chatops development), and it will point traffic directly to your Docker images.
+You can even launch an `ngrok` service locally on your laptop, pointing to port 8080 (such as for chatops development), and it will point traffic directly to your Docker images.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:387]: Same comment about indentation.

File: docs/dev/dev_environment.md [hunk 11]
```diff
@@ -25,7 +25,10 @@ Using **Invoke** these configuration options can be overridden using [several me
 ### Docker Development Environment
 
 !!! tip
-    This is the recommended option for development.
+As of 1.11.1 the Docker Compose environment has moved from the standalone docker-compose to the plugin execution of `docker compose`. Be sure that your version of Docker has the plugin installed. If the docker compose plugin is installed, no other changes to the invoke methodology is required.
+
+!!! tip
+This is the recommended option for development.
```
Comment (smk4664) [inline docs/dev/dev_environment.md:31]: Add indentation for MKDocs admonition.

**Review by smk4664 — CHANGES_REQUESTED** [review]

File: docs/dev/dev_environment.md (hunk 1 above)
Comment (jvanderaa) [reply docs/dev/dev_environment.md:76]: @smk4664 we really need to get the MKDocs admonitions work with standard markdown formatters.

Comment (jvanderaa) [reply docs/dev/dev_environment.md:76]: This is just from a save that had auto format on, which is still my preferred method.

**Comment by jvanderaa** [comment]
The MKDocs stuff should really move to something with `>`, like using `>>>`

**Comment by jvanderaa** [comment]
@smk4664 can we merge this? I just set up a new system and needed to re-implement this change just to get to doing some work.

File: tasks.py [hunk 12]
```diff
@@ -99,6 +99,8 @@ def docker_compose(context, command, **kwargs):
         "PYTHON_VER": context.nautobot_chatops.python_ver,
         **kwargs.pop("env", {}),
     }
+
+    compose_command = f'docker compose --project-name {context.nautobot_chatops.project_name} --project-directory "{context.nautobot_chatops.compose_dir}"'
```
Comment (smk4664) [inline tasks.py:103]: I don't understand this change, it is just going to be overridden a few lines below when it joins the compose command tokens.

**Review by smk4664 — COMMENTED** [review]
It looks like this may not be needed anymore with a previous pr updating this already.

File: tasks.py (hunk 12 above)
Comment (jvanderaa) [reply tasks.py:103]: What is the difference between the tokens and this? It was a merge conflict, and I am not familiar with command tokens.

Comment (smk4664) [reply tasks.py:103]: #212 Replaced `docker-compose` with `docker compose` among other changes. I believe this is why you get the merge conflict.

**Pull request closed without merging**